spec/haskell/src/SEL4/Object/SchedContext.lhs

```diff
@@ -203,20 +201,16 @@ This module uses the C preprocessor to select a target architecture.
 > refillPopHead :: PPtr SchedContext -> Kernel Refill
 > refillPopHead scPtr = do
 >     refill <- liftM refillHd $ getSchedContext scPtr
->     oldHead <- liftM scRefillHead $ getSchedContext scPtr
->     nextHead <- getRefillNext scPtr oldHead
->     updateSchedContext scPtr $ \sc -> sc { scRefillHead = nextHead,
+>     updateSchedContext scPtr $ \sc -> sc { scRefillHead = refillNextIndex (scRefillHead sc) sc,
 >                                     scRefillCount = scRefillCount sc - 1 }
 >     return refill
 
 > refillAddTail :: PPtr SchedContext -> Refill -> Kernel ()
 > refillAddTail scPtr refill = do
 >     sc <- getSchedContext scPtr
->     sz <- getRefillSize scPtr
->     assert (sz < scRefillMax sc) "cannot add beyond queue size"
->     newTail <- getRefillNext scPtr (refillTailIndex sc)
->     refills <- return $ replaceAt newTail (scRefills sc) refill
->     setSchedContext scPtr (sc { scRefills = refills, scRefillCount = sz + 1 })
+>     assert (scRefillCount sc < scRefillMax sc) "cannot add beyond queue size"
+>     newTail <- return $ refillNextIndex (refillTailIndex sc) sc
```

Comment:
I think that this now isn't used
```suggestion
```

Reply:
I think that you might have accidentally added this back in when changing `replaceAt` to `updateAt`.

Reply:
oops, you are right, I will remove it.